drivers/ethernet/eth_stm32_hal.c

```diff
@@ -306,6 +354,69 @@ static int eth_tx(const struct device *dev, struct net_pkt *pkt)
 	}
 #endif /* CONFIG_SOC_SERIES_STM32H7X */
 
+#if defined(CONFIG_PTP_CLOCK_STM32_HAL)
+	if (timestamped_frame) {
+		/* Retrieve transmission timestamp from last DMA TX descriptor */
+#if defined(CONFIG_SOC_SERIES_STM32H7X)
+		ETH_TxDescListTypeDef * dma_tx_desc_list;
+
+		__IO ETH_DMADescTypeDef *last_dma_tx_desc;
```

Comment:
Are you sure __IO needed here? Usually it is STM32 HAL specific macro. Thus I would prefer to use C construct. As I remember __IO should be defined as volatile or so.

Reply:
I kept it around because eth_tx() and eth_rx() both have `__IO ETH_DMADescTypeDef` variables. It might be useful because this memory is populated concurrently with main CPU processing, so CPU shouldn't assume any caching guarantees about it?

Reply:
Well lets leave __IO